src/DotNetty.Buffers/SwappedByteBuffer.cs

```diff
@@ -0,0 +1,608 @@
+            return this;
+        }
+
+        public IByteBuffer ReadBytes(byte[] destination, int dstIndex, int length)
+        {
+            this.buf.ReadBytes(destination, dstIndex, length);
+            return this;
+        }
+
+        public IByteBuffer ReadBytes(Stream destination, int length)
+        {
+            this.buf.ReadBytes(destination, length);
+            return this;
+        }
+
+        public IByteBuffer SkipBytes(int length)
+        {
+            this.buf.SkipBytes(length);
+            return this;
+        }
+
+        public IByteBuffer WriteBoolean(bool value)
+        {
+            this.buf.WriteBoolean(value);
+            return this;
+        }
+
+        public IByteBuffer WriteByte(int value)
+        {
+            this.buf.WriteByte(value);
+            return this;
+        }
+
+        public IByteBuffer WriteShort(int value)
+        {
+            this.buf.WriteShort(ByteBufferUtil.SwapInt(value));
+            return this;
+        }
+
+        public IByteBuffer WriteUnsignedShort(int value)
+        {
+            this.buf.WriteUnsignedShort(ByteBufferUtil.SwapInt(value));
+            return this;
+        }
+
+        public IByteBuffer WriteInt(int value)
+        {
+            this.buf.WriteInt(ByteBufferUtil.SwapInt(value));
+            return this;
+        }
+
+        public IByteBuffer WriteUnsignedInt(uint value)
+        {
+            unchecked
+            {
+                this.buf.WriteUnsignedInt((uint)ByteBufferUtil.SwapInt((int)value));
+            }
+
+            return this;
+        }
+
+        public IByteBuffer WriteLong(long value)
+        {
+            this.buf.WriteLong(ByteBufferUtil.SwapLong(value));
+            return this;
+        }
+
+        public IByteBuffer WriteChar(char value)
+        {
+            this.WriteShort(value);
+            return this;
+        }
+
+        public IByteBuffer WriteDouble(double value)
+        {
+            this.WriteLong(BitConverter.DoubleToInt64Bits(value));
+            return this;
+        }
+
+        public IByteBuffer WriteBytes(IByteBuffer src)
+        {
+            this.buf.WriteBytes(src);
+            return this;
+        }
+
+        public IByteBuffer WriteBytes(IByteBuffer src, int length)
+        {
+            this.buf.WriteBytes(src, length);
+            return this;
+        }
+
+        public IByteBuffer WriteBytes(IByteBuffer src, int srcIndex, int length)
+        {
+            this.buf.WriteBytes(src, srcIndex, length);
+            return this;
+        }
+
+        public IByteBuffer WriteBytes(byte[] src)
+        {
+            this.buf.WriteBytes(src);
+            return this;
+        }
+
+        public IByteBuffer WriteBytes(byte[] src, int srcIndex, int length)
+        {
+            this.buf.WriteBytes(src, srcIndex, length);
+            return this;
+        }
+
+        public bool HasArray { get { return this.buf.HasArray; } }
```

Comment:
Should be reformatted on the new version but I don't know if the diff picks up whitespace - is this issue mostly carriage returns?